### PR TITLE
Add getPriceAtBlock and getPredictedPrice

### PR DESCRIPTION
**Pull request by AddressXception**
Add `getPriceAtBlock` and `getPredictedPrice` methods

## To Test:
1. `yarn test`
2. expected: all tests pass

### CODE REVIEW

File: packages/http/src/models/Balance.ts [hunk 1]
```diff
@@ -22,12 +22,14 @@ export default class Balance {
   public floatBalance: number
   public type: CurrencyType
   public bigBalance: BigNumber
+  public height?: number
 
-  constructor(integerBalance: number | undefined, type: CurrencyType) {
+  constructor(integerBalance: number | undefined, type: CurrencyType, height?: number) {
```
Comment (AddressXception) [inline packages/http/src/models/Balance.ts:27]: i wasn't sure the preferred order here so i just put it at the end

File: packages/http/src/resources/Oracle.ts [hunk 2]
```diff
@@ -10,9 +10,27 @@ export default class Oracle {
...
+    const { data: { data: { price, block } } } = await this.client.get(url)
+
+    return new Balance(price, CurrencyType.usd, block)
+  }
+
+  async getPredictedPrice(): Promise<any> {
+    const url = '/oracle/predictions'
+    const response = await this.client.get(url)
+    if (!response.data.data || !response.data.data.length) {
+      return []
```
Comment (AddressXception) [inline packages/http/src/resources/Oracle.ts:30]: to be consistent with the way the API behaves, returning an empty array.  Could be changed.

Comment (allenan) [reply packages/http/src/resources/Oracle.ts:30]: I think that's fine to stick with how the API behaves

File: packages/http/src/models/Balance.ts (hunk 1 above)
Comment (allenan) [reply packages/http/src/models/Balance.ts:27]: Is it necessary to include `height` within Balance? I know the API returns `time` and `height` depending on the route so there's already a bit of inconsistency there. Maybe we could do like you do in the predicted price route and return a `{ price, height }` object?

Comment (allenan) [reply packages/http/src/models/Balance.ts:27]: That would also have the advantage of conforming more closely with what the API returns

Comment (AddressXception) [reply packages/http/src/models/Balance.ts:27]: i like that, i'll make the change there

**Comment by allenan** [comment]
this looks good, i'm just going to revert out the changes in Balance so that it doesn't create a diff

**Comment by allenan** [comment]
actually I might need you to make that change @AddressXception. Could you revert out the comma changes in Balance.ts so that it doesn't create unnecessary diffs? Then I'll get this merged and published for you

**Pull request merged**